spacy_layout/layout.py

```diff
@@ -112,19 +124,42 @@ def _texts_to_doc(self, inputs: list[tuple[str, str, SpanLayout]]) -> Doc:
                     spaces[-1] = False
                     spaces.append(False)
                 end = token_idx + len(span_doc)
-                span_data.append((token_idx, end, label, layout))
+                span_data.append((item, token_idx, end))
                 token_idx += len(span_doc) + (1 if self.sep else 0)
         doc = Doc(self.nlp.vocab, words=words, spaces=spaces)
         spans = []
-        for i, (start, end, label, layout) in enumerate(span_data):
-            span = Span(doc, start=start, end=end, label=label, span_id=i)
+        for i, (item, start, end) in enumerate(span_data):
+            span = Span(doc, start=start, end=end, label=item.label, span_id=i)
+            layout = self._get_span_layout(item, pages)
             span._.set(self.attrs.span_layout, layout)
+            if item.label == DocItemLabel.TABLE:
+                span._.set(self.attrs.span_data, item.export_to_dataframe())
             spans.append(span)
         doc.spans[self.attrs.span_group] = SpanGroup(
             doc, name=self.attrs.span_group, spans=spans
         )
         return doc
 
+    def _get_span_layout(
+        self, item: DoclingItem, pages: dict[int, PageLayout]
+    ) -> SpanLayout | None:
+        bounding_box = None
+        if item.prov:
+            prov = item.prov[0]
+            page = pages[prov.page_no]
+            if page.width and page.height:
+                is_bottom = prov.bbox.coord_origin == CoordOrigin.BOTTOMLEFT
+                y = page.height - prov.bbox.t if is_bottom else prov.bbox.t
+                height = prov.bbox.t - prov.bbox.b if is_bottom else prov.bbox.t
```

Comment:
Just flagging this one up because I always find this sort of stuff tricky. It might be worth moving just this maths to a function so you can test it more easily. If you're confident it's all correct nevermind.

Reply:
Yeah, this took me a while to get right, even though it's logic I've written many times 😅 I am pretty confident about it now and it's tested but I can go and add more explicit tests later just to be sure.